### PR TITLE
NTP packet generation and parsing demo

### PR DESCRIPTION
**Pull request by FedericoCeratto**
A little demo

### CODE REVIEW

File: demos/ntp.ntp [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+
+proc main() =
+  if paramCount() != 1:
+    echo "Usage $# <ipaddr>" % paramStr(0)
+    quit(1)
+
+  let target = paramStr(1)
+  var b = NTPPacket()
+
+  # example values
```
Comment (FedericoCeratto) [inline demos/ntp.ntp]: I think we can move the discussion to a dedicated issues. Anyhow, as you are saying, a serialization proc should not care about semantics.
A fill() procedure or even a dedicated library could support default values (fixed or random) and maybe callbacks to perform more complex generation e.g. a checksum based on other fields.

File: demos/ntp.ntp [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+  b.stratum = 0x03
+  b.polling_interval = 0x06
+  b.clock_precision = 0xfa
+  b.delay = 0x00010000
+  b.dispersion = 0x00010000
+  b.reference_id = 0xaabbcc
+  b.origin_ts = epochTime().epoch_to_ntp_ts
+
+  var bytes = ""
+  for i in b.serialize():
```
Comment (FedericoCeratto) [inline demos/ntp.ntp]: Updated, thanks.
I imagine you are planning to add a serialize() that simply returns a byte string?

File: demos/ntp.ntp [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+    bytes.add chr(i)
+
+  assert bytes.len == 48
+
+  var socket = newSocket(AF_INET, SOCK_DGRAM, IPPROTO_UDP, true)
+  # client send time
+  let t1 = epochTime()
+  doAssert socket.sendTo(target, Port(123), bytes) == 48
+
+  let raw_resp = socket.recv(48, timeout=1000)
```
Comment (FedericoCeratto) [inline demos/ntp.ntp]: Updated.

File: demos/ntp.ntp (hunk 2 above)
Comment (xomachine) [inline demos/ntp.ntp]: https://github.com/xomachine/NESM/commit/c3f02321da8fecdddf73c4a95980b6a8c141eade has introduced the Stream interface in the "devel" branch. Now you can use StringStream for working with strings. This commit is also changed `serialize()`'s return value from `seq[byte]` to `string`. The "devel" branch will be merged into master after completion of implementing endians support and some documentation rework.

**Comment by xomachine** [comment]
Merged to devel with changes via e1fa6d87204f2b41c0858e7425d7435c68b724bb and 0da16e2eaf59e92b022e36eb81a7c227b7071c68

**Pull request closed without merging**